### DISCUSSION TITLE
Product in mutilple category

### DISCUSSION THREAD

**Discussion by balanh-development** [swellstores/community#146] [Help] [answered]
Hi Swell Team,

I follow your docs in link below
https://developers.swell.is/backend-api/querying/filtering

I have few questions: I have to custom some pages for speciall business. Is there any rest API support add product to mutilple category ? More over, I want to retrieve product from mutilple category, is swell rest support it ?

Filtering like Mongo how about $In term and others complex filter ?

Event I look for in https://developers.swell.is/frontend-api/products#list-products-by-category
It's simple use case, in the real world, our customer want to search product combine mutilple condition ex: size, color, name, category. How we do that ?

**Comment by sags95** [comment, marked as answer]
Hi @balanh-development , 

You can use the `/categories:products` endpoint in order to assign the product to categories. You would pass in the category id as the `parent_id`.
```
POST /categories:products { product_id, parent_id }
```

To retrieve products from multiple categories, you can pass in a list of categories:
```
GET '/products', {
  categories: [...ids]
}
```

As to your question regarding filtering on the frontend, this would be done by passing in attributes within the `$filters` object like you describe (size, color, etc) along with the category.
```
await swell.products.list({
  page,
  limit,
  sort,
  categories, // optional category slug or ID
  $filters: {
    price: [10, 20],
    category: ['featured'],
    size: ['small', 'large']
  }
})
```

**Reply by dyk9456** [reply]
How can we do the filtering in front end using graphQl

**Reply by logeshswell** [reply]
Please refer to the below documentation on `graphql` library:
https://developers.swell.is/frontend-api/products#list-products

```graphql
query getAllProducts {
  products(limit: 25, page: 1, where: { stock_level: 0 }) {
    results {
      id
      name
      slug
      price
      currency
      options {
        id
        attributeId
        name
        inputType
        active
        required
        variant
        values {
          id
          name
          price
          description
        }
      }
      purchaseOptions {
        standard {
          price
          sale
          salePrice
        }
        subscription {
          plans {
            id
            name
            price
            billingSchedule {
              interval
              intervalCount
            }
          }
        }
      }
    }
  }
}
```

The accepted parameters for this query are search, where, sort, limit, page, _locale, _currency, and _pricing.
